pyparsing/core.py

```diff
@@ -297,21 +297,20 @@ def wrapper(*args):
                 # re-raise TypeErrors if they did not come from our arity testing
                 if found_arity:
                     raise
-                else:
```

Comment:
Please move this to a separate PR. _trim_arity is a touchy piece of code so should be changed in isolation from other changes.